src/InputstreamDemux.cpp

```diff
@@ -386,8 +386,11 @@ void cVNSIDemux::StreamChange(cResponsePacket* resp)
       const char* language = resp->extract_String();
       uint32_t composition_id(resp->extract_U32());
       uint32_t ancillary_id(resp->extract_U32());
-      props.SetLanguage(language);
-      props.SetSubtitleInfo((composition_id & 0xffff) | ((ancillary_id & 0xffff) << 16));
+      if (std::strlen(language) > 0)
```

Comment:
Well, looking at the implementation of `PVRStreamProperties::SetLanguage` (https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/pvr/Stream.h#L157) we learn that it expects a 3-letter language id, checks for string not longer than 3 letters, but has no check for string length zero, 1 or 2. So, passing a string with a length less than 3 will lead to a crash. 

@AlwinEsch I think you should fix the implementation of `PVRStreamProperties::SetLanguage`
@gipawu the right workaround for the above mentioned bug in Kodi core is `if (std::strlen(language) != 3)`